File: app/src/main/java/com/example/test/MainActivity.java
package com.example.test;


import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView tv = findViewById(R.id.textview);

        Button btn = findViewById(R.id.btn1);
        btn.setOnClickListener((v) -> {
            tv.setText(NativeUtils.getCString());
        });
    }

    @Override
    protected void onResume() {
        super.onResume();
    }
}

File: app/src/main/java/com/example/test/biz1/MainActivity.kt
package com.example.test.biz1

import android.app.appfunctions.AppFunctionManager
import android.content.Context
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.test.R

class MainActivity: AppCompatActivity() {
    companion object {
        const val TAG = "MainActivityTAG"
    }
    private lateinit var mLogView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mLogView = findViewById(R.id.logTv)

        findViewById<Button>(R.id.envBtn).setOnClickListener {
            checkEnv()
        }
    }

    private fun checkEnv() {
        try {
            val mgr = getSystemService(Context.APP_FUNCTION_SERVICE) as AppFunctionManager
            setLog("mgr=$mgr")
        } catch (e: Exception) {
            Log.e(TAG, "${e.message}", e)
            mLogView.text = "${e.message}"
        }
    }

    private fun setLog(text: String?) {
        text?.let {
            mLogView.post { mLogView.text = it }
        }
    }

    private fun addLog(text: String?) {
        text?.let {
            mLogView.post { mLogView.text = it + mLogView.text.toString() }
        }
    }
}
